### DISCUSSION TITLE
Enhancement to ASG filters

### DISCUSSION THREAD

**Discussion by ndesai15** [cloud-custodian/cloud-custodian#7347] [General]
I have come across one situation when ASG has been created with Launch Template & Launch template is tag compliance & propagating tags to instances. Now I have a policy for ASG tag validation. It is same as ASG tag compliance example policy, provided in custodian documentation. This policy checks only at ASG level (propagated-tags filter also just checks for if a list of tags have been propagated or not from ASG. It doesn't check instance level tags attached to ASG).This is creating a problem. Even though my instances are being tagged properly & ASG is not-tagged properly, it is marking ASG as non-compliant. But ASG is not costing money instances, under-hood ASG costing money.

**Comment by kapilt** [comment]
if you have a tag policy on the asg thats verifying your tag regime with propagation checks then it will eventually go to all instances, you can have a separate policy wrt to reporting on extant ec2, or you can do a propagate action that will copy compliant asg tags to instances as a one off policy activity.

**Comment by ndesai15** [comment]
Do you mean propagate-tags filter? This filter is designed to find all asgs that have a list of tag keys that are set to propagate to new instances. But I am talking about what is ASG doesn't have tag & instances that are part of ASG have all proper tags via lunch-template? How can I check in this case?
